--- libq/colors.c.orig	2018-03-26 11:21:13.063768978 +0300
+++ libq/colors.c	2018-03-26 11:24:28.749778309 +0300
@@ -21,10 +21,11 @@
 static const char *COLOR_MAP = CONFIG_EPREFIX "etc/portage/color.map";
 
 #define COLOR _MAKE_COLOR
+#define CPAIR_VALUE_LEN 16
 
 typedef struct {
 	const char *name;
-	char value[16];
+	char value[CPAIR_VALUE_LEN];
 } cpairtype;
 
 static cpairtype color_pairs[] = {
@@ -42,6 +43,7 @@
 	{"turquoise", COLOR("36", "01") },
 	{"yellow",    COLOR("01", "33") },
 	{"white",     COLOR("01", "38") },
+	{"lightgray", COLOR("00", "37") },
 	{"eol",       COLOR("00", "00") },
 };
 
@@ -77,7 +79,22 @@
 				if (strncmp(p, "0x", 2) == 0)
 					warn("[%s=%s] RGB values in color map are not supported on line %d of %s", buf, p, lineno, COLOR_MAP);
 				else
-					snprintf(color_pairs[i].value, sizeof(color_pairs[i].value), "\e[%s", p);
+				{
+					/* color=color format support */
+					int n, found = 0;
+					for (n = 0; n < ARRAY_SIZE(color_pairs); n++)
+					{
+						if (strcmp(color_pairs[n].name, p) == 0)
+						{
+							strncpy(color_pairs[i].value, color_pairs[n].value, CPAIR_VALUE_LEN); 
+							found = 1;
+							break;
+						}
+					}
+
+					if (!found)
+						snprintf(color_pairs[i].value, sizeof(color_pairs[i].value), "\e[%s", p);
+				}
 			}
 	}
 
